src/network/ClientConnectionManager.ts

```diff
@@ -106,17 +108,213 @@ enum ClientState {
     INITIALIZED_ON_CLUSTER = 2,
 }
 
+export interface ConnectionRegistry {
+    /**
+     * Returns if the registry active
+     * @return Return true if the registry is active, false otherwise
+     */
+    isActive(): boolean;
+
+    /**
+     * Returns if the registry empty
+     * @return true if the registry empty, false otherwise
+     */
+    isEmpty(): boolean;
+
+    /**
+     * Returns connection by UUID
+     * @param uuid UUID that identifies the connection
+     * @return ClientConnection if there is a connection with the UUID, undefined otherwise
+     */
+    getConnection(uuid: UUID): ClientConnection | undefined;
+
+    /**
+     * Returns all active connections in the registry
+     * @return Array of ClientConnection objects
+     */
+    getConnections(): ClientConnection[];
+
+    /**
+     * Returns a random connection from active connections
+     * @return ClientConnection if there is at least one connection, otherwise null
+     */
+    getRandomConnection(): ClientConnection | null;
+
+    /**
+     * Runs the given function for every active connection
+     * @param fn A function that takes ClientConnection as arg, does not return anything
+     */
+    forEachConnection(fn: (conn: ClientConnection) => void): void;
+
+    /**
+     * Returns if invocation allowed. Invocation is allowed only if connection state is {@link INITIALIZED_ON_CLUSTER}
+     * and there is at least one active connection.
+     * @return Error if invocation is not allowed, null otherwise
+     */
+    checkIfInvocationAllowed(): Error | null;
+
+    /**
+     * Delete connection from active connections
+     * @param uuid UUID that identifies the connection
+     */
+    deleteConnection(uuid: UUID): void;
```

Comment:
This interface should not have any mutation methods as their only consumer is `ClientConnectionManager` which uses the implementation class.

Reply:
Currently, connection registry is passed to connection manager with interface `ConnectionRegistry`. Does it mean I should change that to `ConnectionRegistryImpl`

Reply:
Also, should I remote non mutating methods that only connection manager uses?

Reply:
> Does it mean I should change that to `ConnectionRegistryImpl`

Yes. In future we could make connection manager the owner of this object (so that it creates it and provides the client with `ConnectionRegistry` for further initialization), but for now it's fine to pass the concrete class.

> Also, should I remove non mutating methods that only connection manager uses?

Yes, it makes sense to expose only methods used by the consumers.